test/blackbox-tests/test-cases/hidden-deps.t/run.t

```diff
@@ -0,0 +1,36 @@
+Should include Foo with -H:
+
+  $ getincludes () {
+  > dune build --verbose ./run.exe 2>&1 | grep run.ml | grep -E -o "\-$1\s(.foo)\S*" | sed s/\-$1//g | tr -d '[:space:]'
+  > }
+
+  $ supports_H=$(if ocamlc -H x --help >/dev/null 2>&1; then echo true; else echo false; fi)
```

Comment:
I would recommend doing that in 2 tests:
- one that assumes ocaml >= 5.2 (guarded with `(enabled_if (>= %{ocaml_version} 5.2))`)
- one that assumes ocaml < 5.2 (similarly guarded)

And then you can use "normal" cram tests, which would consist in just running `getincludes` and displaying the output.

Reply:
Hello, thanks for the review.
I am familiar with gaurding executables and libraries with the `enabled_if` (essentially making them hidden to the test). However, this would not be very useful here.

I have tried to add the `enabled_if` to a `cram` stanza in the dune file, hoping that would disable the whole test. This seems to not work since when I try to build the test it still runs irrespective of the condition mentioned with `enabled_if`.

Do you have any suggestions on how one would use `enabled_if` in order to disable the whole test?

Reply:
sure! you can have a look at this for example:

https://github.com/ocaml/dune/blob/3.16.0/test/blackbox-tests/test-cases/dune#L103-L107